llvm/lib/SYCLLowerIR/SanitizeDeviceGlobal.cpp

```diff
@@ -43,12 +43,21 @@ static bool instrumentDeviceGlobal(Module &M) {
   //  size_t beginning address of the device global
   StructType *StructTy = StructType::get(IntTy, IntTy, IntTy);
 
+  Metadata *OMPDeviceMD = M.getModuleFlag("openmp-device");
   for (auto &G : M.globals()) {
-    // Non image scope device globals are implemented by device USM, and the
-    // out-of-bounds check for them will be done by sanitizer USM part. So we
-    // exclude them here.
-    if (!isDeviceGlobalVariable(G) || !hasDeviceImageScopeProperty(G))
-      continue;
+    if (OMPDeviceMD) {
+      // For openmp offloading, the scope is to check external global variables
+      // within '#pragma declare target' block and ignore all internal names.
+      if (G.getName().starts_with("__") || G.getName().starts_with(".") ||
+          !G.hasExternalLinkage())
```

Comment:
We should probably swap the order of checks to first check for linkage and then for the name to make this `if` more efficient

Reply:
Hi, @AlexeySachkov 
Done.
Thanks very much.